app/models/search/bookmark_query.rb

```diff
@@ -140,10 +143,19 @@ def aggregations
   def visibility_filters
```

Comment:
Just checking: we don't need the `posted_filter` here because this is used for _bookmarks_, and only _bookmarkables_ can be unposted?

Reply:
Roughly, yeah. The `posted_filter` in the `BookmarkQuery` class has always been a parent filter (constructed with the `parent_term_filter` function), meaning it applies only to the bookmarkable. The old `posted_filter` was included in the AND/`must` portion of the query, meaning that there was a constraint in the query saying, "in order to return this bookmark, it must have a bookmarkable that has `posted` set to `true`." However, the "it must have a bookmarkable" part of that constraint meant that that query could never return any bookmarks with a deleted bookmarkable.

The idea behind this pull request is to take that constraint and take the double negative: "in order to return this bookmark, it must *NOT* have a bookmarkable that has `posted` set to `false`." That way it will correctly exclude bookmarks on draft works, correctly allow bookmarks on posted works, and correctly allow bookmarks on deleted works.

To implement that modification to the constraint, I moved `posted_filter` from `visibility_filters` (which are included in the AND/`must` portion of the query) to `bookmarkable_visibility_filters` (which are included in the NOT/`must_not` portion of the query). I also flipped the boolean from "true" to "false" to get the proper double-negative.

(I also renamed `posted_filter` to `unposted_parent_filter`, on the theory that the name should reflect its role better. But I think that may have been more confusing in the short term.)

Reply:
`unposted_parent_filter` totally made sense to me, fwiw -- it was kind of my "ah-ha, I see" moment, in fact. I think it's really easy to understand "we're just excluding things that are marked unposted, so everything else gets included."